sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/MethodsShouldNotHaveIdenticalImplementations.cs

```diff
@@ -51,12 +51,17 @@ protected override IEnumerable<MethodBlockSyntax> GetMethodDeclarations(SyntaxNo
         protected override bool AreDuplicates(MethodBlockSyntax firstMethod, MethodBlockSyntax secondMethod)
         {
             return firstMethod.Statements.Count >= 2 &&
-                firstMethod.SubOrFunctionStatement.Identifier.ValueText != secondMethod.SubOrFunctionStatement.Identifier.ValueText &&
-                HaveSameParameters(firstMethod?.BlockStatement?.ParameterList.Parameters, secondMethod?.BlockStatement?.ParameterList.Parameters) &&
-                VisualBasicEquivalenceChecker.AreEquivalent(firstMethod.Statements, secondMethod.Statements);
+                   firstMethod.GetIdentifierText() != secondMethod.GetIdentifierText() &&
+                   HaveSameParameters(firstMethod.BlockStatement?.ParameterList.Parameters, secondMethod.BlockStatement?.ParameterList.Parameters) &&
+                   VisualBasicEquivalenceChecker.AreEquivalent(firstMethod.Statements, secondMethod.Statements);
 
             bool HaveSameParameters(SeparatedSyntaxList<ParameterSyntax>? leftParameters, SeparatedSyntaxList<ParameterSyntax>? rightParameters)
             {
+                if (leftParameters == null && rightParameters == null)

```

Comment:
Since we cannot reproduce this issue and I don't have the full stack-trace I had to do a bit of guess work here. We seem to handle the cases when only one of the parameters (left or right) is null but not when both are null. I assume that this might be the reason why we get the exception but I'm not sure yet.

Reply:
From [the docs ](https://docs.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/procedures/sub-procedures), the declaration syntax for Sub is `[ modifiers ] Sub subname [( parameterlist )]`

from what I understand, it means the `()` are optional, so you can add tests in [MethodsShouldNotHaveIdenticalImplementations.vb](https://github.com/SonarSource/sonar-dotnet/blob/c6abd066b90df6d5b9f2fe484fb21d61ea404a30/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodsShouldNotHaveIdenticalImplementations.vb) where you have some subs like

```
        Sub FooNoParens1
            Dim s As String = "test"
            Console.WriteLine("Result: {0}", s)
        End Sub

        Sub FooNoParens2
            Dim s As String = "test"
            Console.WriteLine("Result: {0}", s)
        End Sub
```

ideally this should solve the coverage problem

Reply:
Nice, That's what I was missing. I've updated the code.